### PR TITLE
feat(metrics): add module rejection and fallback gauges

### PR DESCRIPTION
**Pull request by JAdshead**
## Description

added two additional holocron gauges. 

1. to track the total number of rejected modules after each module map update.
2. to track the total number of modules which are using fallback externals.

## Motivation and Context

Provide a high level metrics to indicate to users how unhealthy the state of their module map is.
Information on rejected modules and modules using externals information is already logged so this feature does not provide insight where there was none but exposes it in a different light.

## How Has This Been Tested?

Locally started one-app with both modules which would be rejected and ones which load but required external fallbacks.

## Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (adding or updating documentation)
- [ ] Dependency update
- [ ] Security update

## Checklist:

- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] These changes should be applied to a maintenance branch.
- [ ] This change requires cross browser checks.
- [ ] Performance tests should be ran against the server prior to merging.
- [ ] This change impacts caching for client browsers.
- [ ] This change impacts HTTP headers.
- [ ] This change adds additional environment variable requirements for One App users.
- [ ] I have added the Apache 2.0 license header to any new files created.

### CODE REVIEW

File: __tests__/server/utils/pollModuleMap.spec.js [hunk 1]
```diff
@@ -284,11 +279,11 @@ describe('pollModuleMap', () => {
...
 
-    expect(setGauge).toHaveBeenCalledTimes(1);
-    expect(setGauge.mock.calls[0][0]).toBe(holocronMetrics.moduleMapPollWait);
-    expect(setGauge.mock.calls[0][1]).toBeGreaterThanOrEqual(5 * 1.25);
+    expect(setGauge).toHaveBeenCalledTimes(3);
+    expect(setGauge.mock.calls[2][0]).toBe(holocronMetrics.moduleMapPollWait);
+    expect(setGauge.mock.calls[2][1]).toBeGreaterThanOrEqual(5 * 1.25);
     // ensure we're using seconds, not milliseconds
-    expect(setGauge.mock.calls[0][1]).toBeLessThan(8 * 1.25);
+    expect(setGauge.mock.calls[2][1]).toBeLessThan(8 * 1.25);
```
Comment (dogpatch626) [inline __tests__/server/utils/pollModuleMap.spec.js:286]: why are we multiplying by 1.25?

Comment (JAdshead) [reply __tests__/server/utils/pollModuleMap.spec.js:286]: we increase the poll time by `*1.25` each cycle - 
https://github.com/americanexpress/one-app/pull/1131/files#diff-aee22318f291692511a5103c699287cfeb2d61fbc7b3ae9d8395019a73c947d9R62

File: src/server/utils/loadModules.js [hunk 2]
```diff
@@ -39,6 +39,7 @@ const loadModules = async () => {
   }
   cachedModuleMapHash = moduleMapHash;
   const serverConfig = getServerStateConfig();
+  // NODE:: this function mutates the moduleMap
```
Comment (Matthew-Mallimo) [inline src/server/utils/loadModules.js:42]:
```suggestion
  // NOTE:: this function mutates the moduleMap
```
?

**Review by Matthew-Mallimo — DISMISSED** [review]

File: src/server/metrics/holocron.js [hunk 3]
```diff
@@ -46,4 +46,14 @@ createGauge({
...
 });
 
+createGauge({
+  name: holocronNamespace('rejected_modules', 'total'),
+  help: 'how many holocron modules have failed to load',
+});
+
+createGauge({
+  name: holocronNamespace('modules_requiring_fallbacks', 'total'),
+  help: 'how many modules which require external fallbacks',
```
Comment (PixnBits) [inline src/server/metrics/holocron.js:56]: #nitpick :
```suggestion
  help: 'how many Holocron modules which require external fallbacks',
```

File: src/server/metrics/holocron.js [hunk 4]
```diff
@@ -46,4 +46,14 @@ createGauge({
   help: 'how many times has the server encountered errors polling the module map',
 });
 
+createGauge({
+  name: holocronNamespace('rejected_modules', 'total'),
+  help: 'how many holocron modules have failed to load',
```
Comment (PixnBits) [inline src/server/metrics/holocron.js:51]: #nitpick :
```suggestion
  help: 'how many Holocron modules have failed to load',
```

File: __tests__/server/utils/pollModuleMap.spec.js [hunk 5]
```diff
@@ -328,7 +323,7 @@ describe('pollModuleMap', () => {
     // check that the poll interval is reset to min.
     expect(setTimeout).toHaveBeenCalledWith(pollModuleMap, MIN_POLL_TIME);
 
-    expect(setGauge).toHaveBeenCalledTimes(1);
+    expect(setGauge).toHaveBeenCalledTimes(3);
     expect(setGauge).toHaveBeenCalledWith(holocronMetrics.moduleMapPollWait, 5);
```
Comment (PixnBits) [inline __tests__/server/utils/pollModuleMap.spec.js:327]: should this include
```suggestion
    expect(setGauge).toHaveBeenCalledWith(holocronMetrics.moduleMapPollWait, 5);
    expect(setGauge).toHaveBeenCalledWith(holocronMetrics.rejectedModules, Infinity);
    expect(setGauge).toHaveBeenCalledWith(holocronMetrics.modulesRequiringFallbacks, Infinity);
```
alongside the `.toHaveBeenCalledTimes(` assertion updates?

File: __tests__/server/utils/pollModuleMap.spec.js [hunk 6]
```diff
@@ -231,6 +222,10 @@ describe('pollModuleMap', () => {
     expect(setTimeout).toHaveBeenCalledTimes(2);
     expect(setTimeout.mock.calls[1][0]).toBe(pollModuleMap);
 
+    setGauge.mockImplementationOnce(() => {
+      throw new Error('unable to increment gague');
```
Comment (PixnBits) [inline __tests__/server/utils/pollModuleMap.spec.js:226]: #nitpick
```suggestion
      throw new Error('unable to change gauge');
```
gauges go up and down, _counters_ are incremented 😅

**Review by PixnBits — CHANGES_REQUESTED** [review]
three nitpicks, but one blocker? suggestion

File: __tests__/server/utils/pollModuleMap.spec.js (hunk 5 above)
Comment (JAdshead) [reply __tests__/server/utils/pollModuleMap.spec.js:327]: added below

**Review by 10xLaCroixDrinker — APPROVED** [review]

**Review by Matthew-Mallimo — APPROVED** [review]

File: __tests__/server/utils/pollModuleMap.spec.js (hunk 5 above)
Comment (PixnBits) [reply __tests__/server/utils/pollModuleMap.spec.js:327]: Do we need this in all the tests looking at the telemeters? 🤔

Comment (JAdshead) [reply __tests__/server/utils/pollModuleMap.spec.js:327]: each telemeter has a specific test already see https://github.com/americanexpress/one-app/pull/1131/files#diff-82cec72196769195fc18f858b882683ecae08a9f5fc3b7d097fae4f60d33b404R330-R331
which is why i didn't include the other metrics in this test

**Review by PixnBits — APPROVED** [review]

**Pull request merged**